### PR TITLE
meraki_alert - Fix some alert notifications so they don't crash

### LINKED ISSUE #414: meraki_alert - Default/Alert Destination: email is not optional after initial push
`meraki_alert`'s `alerts:` and `default_destinations` will successfully push configuration without a `emails:` on the initial deployment. If you rerun the task though, it will fail. This prevents you from reusing the playbook. 

You shouldn't need an email on every alert because we expect the default destination to take care of this. Secondly, we do not use emails, we use webhooks for these aka `httpServerIds`

These two tasks can obviously be one task but broken up to show the errors easier that is happens to both sections.
```
    - name: Apply Default Settings for Alerts
      cisco.meraki.meraki_alert:
        auth_key: abc12345
        org_name: YourOrg
        net_name: MyNet
        state: present
        default_destinations:
          all_admins: no
          snmp: no
          #emails: "djohnson@company.com"
          http_server_ids: "xxxx" #webhook id

    - name: Apply Alerts
      cisco.meraki.meraki_alert:
        auth_key: abc12345
        org_name: YourOrg
        net_name: MyNet
        state: present
        alerts:
          - alert_type: "settingsChanged"
            enabled: yes
            alert_destinations:
              all_admins: no
              snmp: no
              #emails: "djohnson@company.com"
      delegate_to: localhost
```

Error when it is ran a second time:
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: object of type 'NoneType' has no len()
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/Users/djohnson/.ansible/tmp/ansible-tmp-1673463346.012788-91328-183734641598206/AnsiballZ_meraki_alert.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/Users/djohnson/.ansible/tmp/ansible-tmp-1673463346.012788-91328-183734641598206/AnsiballZ_meraki_alert.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/Users/djohnson/.ansible/tmp/ansible-tmp-1673463346.012788-91328-183734641598206/AnsiballZ_meraki_alert.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.meraki.plugins.modules.meraki_alert', init_globals=dict(_module_fqn='ansible_collections.cisco.meraki.plugins.modules.meraki_alert', _modlib_path=modlib_path),\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/modules/meraki_alert.py\", line 356, in <module>\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/modules/meraki_alert.py\", line 333, in main\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/module_utils/network/meraki/meraki.py\", line 185, in is_update_required\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/module_utils/network/meraki/meraki.py\", line 176, in is_update_required\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/module_utils/network/meraki/meraki.py\", line 185, in is_update_required\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/module_utils/network/meraki/meraki.py\", line 185, in is_update_required\n  File \"/var/folders/80/qxqt__d92fz5dknpxl46xk9n02zpz9/T/ansible_cisco.meraki.meraki_alert_payload_p6erowjc/ansible_cisco.meraki.meraki_alert_payload.zip/ansible_collections/cisco/meraki/plugins/module_utils/network/meraki/meraki.py\", line 171, in is_update_required\nTypeError: object of type 'NoneType' has no len()\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Adding the email address to the alert destination and the module works as it should. Email address is only optional during initial deployment.

I'm going to guess that it is because `emails` does not have a default of `[]` like `http_server_ids` does maybe. Doesn't explain why it works on the initial push though.

### PR DESCRIPTION
**Pull request by kbreit**
Fixes #414

### CODE REVIEW

**Comment by DerrickTJ** [comment]
If you do not specify a http_server_ids in the task, it will remove it from the site as if a default [] was given. Emails do not do this and stay in the configurations as expected. Does this in both default destinations and alert destinations.

Everything else appears to be working as expected.

**Comment by kbreit** [comment]
They absolutely should operate the same. Let me look into why they're not and also improve my test cases.

**Comment by kbreit** [comment]
I'm building out the test suite a little more to specifically test the emails and HTTP server IDs in multiple situations.

**Comment by kbreit** [comment]
@DerrickTJ Perhaps you can help me here. I'm trying to test the HTTP IDs and when I pass in a base64 webhook URL of `aHR0cHM6Ly93ZWJob29rLnNpdGUvZjExMzMxM2UtZmJhZC00NjRkLTlmYmUtZWQyMzkyZTFlZThjCg==` it isn't applying at all. Just ignores it. Here's my task. It feels like this should work. This happens in both a RESTful API client (RapidAPI) as well as in Ansible.

```
  - name: Set default destination email address and HTTP server ID
    cisco.meraki.meraki_alert:
      auth_key: '{{ auth_key }}'
      org_name: '{{ test_org_name }}'
      net_name: IntTestNetworkWireless
      state: present
      default_destinations:
        all_admins: false
        snmp: false
        emails:
          - '{{ email_prefix }}@{{ email_domain }}'
          - '{{ email_prefix }}2@{{ email_domain }}'
        http_server_ids:
          - "aHR0cHM6Ly93ZWJob29rLnNpdGUvZjExMzMxM2UtZmJhZC00NjRkLTlmYmUtZWQyMzkyZTFlZThjCg=="
    delegate_to: localhost
    register: set_default_lists
```

**Comment by DerrickTJ** [comment]
I'm not passing it through as a list in my test. Since it is as single item, I just put it in as such, without the list format.

**Comment by kbreit** [comment]
> I resolved this problem. I wasn't creating a webhook in advance so there was nothing for it to accept. I'm going to look into the problems you described.

**Comment by kbreit** [comment]
@DerrickTJ Would you please test this PR again? I believe the problems are resolved.

**Comment by DerrickTJ** [comment]
Old playbooks would have expected a default value of nothing. So likely it wasn't even defined in their playbooks. And if it wasn't defined, it would have removed all http_server_ids in the dashboard. This version, it actually does nothing. If anything, this is better.

Dashboard doesn't input a default http_server_ids, so I would think you'd be fine to push it as is. Just my two cents.

**Comment by DerrickTJ** [comment]
My tests all passed.

**Review by DerrickTJ — APPROVED** [review]

File: tests/integration/targets/meraki_alert/tasks/main.yml [hunk 1]
```diff
@@ -1,172 +1,345 @@
...
+        org_name: '{{ test_org_name }}'
+        net_name: IntTestNetworkWireless
+        state: present
+        default_destinations:
+          emails:
+            - '{{ email_prefix }}@{{ email_domain }}'
+            - '{{ email_prefix }}2@{{ email_domain }}'
+          all_admins: false
+          snmp: false
+          http_server_ids:
```
[the remark this answers is not in the document]
Comment (kbreit) [reply tests/integration/targets/meraki_alert/tasks/main.yml:30]: Good catch. Making that change now and will have it uploaded soon.

**Comment by kbreit** [comment]
> Old playbooks would have expected a default value of nothing. So likely it wasn't even defined in their playbooks. And if it wasn't defined, it would have removed all http_server_ids in the dashboard. This version, it actually does nothing. If anything, this is better.
> 
> Dashboard doesn't input a default http_server_ids, so I would think you'd be fine to push it as is. Just my two cents.

It's also the emails that I'm worried about breaking changes for as well. However, I don't think anything I did will break it. Once again, likely improve.

**Pull request merged**